src/Storages/StorageView.cpp

```diff
@@ -112,7 +112,14 @@ StorageView::StorageView(
     : IStorage(table_id_)
 {
     StorageInMemoryMetadata storage_metadata;
-    storage_metadata.setColumns(columns_);
+    if (!is_parameterized_view_)
+    {
+        if (!query.isParameterizedView())
```

Comment:
This looks a bit strange, shouldn't we expect that `is_parametrized_view <=> query.isParametrizedView()`?

Reply:
`isParametrizedView()` is only for CREATE queries, not SELECT. But I get your point, the names are confusing. Will update

Reply:
Added comment, Hope this clarifies